rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+The main downsides of putting this logic in Glides are:
+
+1. As Glides will be more directly responsible for data that flows more directly into passenger-facing systems, it is possible that the Glides team will have to respond to more feedback about issues related to why Green Line trains were assigned to incorrect trips. However, this is likely going to happen anyway with trainsheet-based predictions in RTR given that that already relies on input from Glides.
+1. Glides does not currently have a way of getting location data on yard pull-out / pull-back trains from RTR, so we will need to set up a pipeline for that in order for Glides to implement the full functionality. However, we already need to do this to enable other functionality in Glides, so it is not introducing net new work.
+1. Glides will need to generate automatically-created trips for trains in service that aren't matched to any trip on a trainsheet, logic that is currently implemented in RTR. However, this is likely desirable anyway because Glides will need a place to associate things like trip notes on unmatched trains in the future.
+1. Glides may not be aware of some disruptions in its schedule data that are modeled at the `gtfs_creator` level, meaning that we will still need RTR to create ADDED trips for any scheduled trips that it doesn't recognize. With the current logic, RTR will at least attempt to match to trips in the disrupted GTFS. However, given that these trips aren't guaranteed to match actual operational practice, it's unclear how much value there is in matching to them anyway.
+1. It introduces a very slight lag in trip assignments making their way into passenger-facing data when updated due to train movements. RTR will need to receive the updated location information and Glides will need to receive it, and then Glides will in turn need to make the assignment and transit that information to RTR via an event. However, this added delay is likely minimal, especially if we move towards an event-based architecture for sending vehicle position updates from RTR to Glides (out of scope for this RFC but also something I want to consider in the future).
+
+## Alternative event schema approaches
+
+The approach outlined to changing the realtime event schemas leans strongly towards maintaining an incremental state transfer approach rather than sending all of the data on a given trip with each update. The rationale for this is that RTR needs to take action to update its own state when it receives information from Glides, and sending a full set of data actually makes it harder for RTR to tell exactly what has changed. It is also conceptually similar to the current OCS message architecture, so maintains a sort of symmetry between different data sources that RTR handles. This should, however, be flagged as an item for discussion in case these assumptions are incorrect.
```

Comment:
Is this referring to `trip_updated v1`, `trip_updated v2`, or `trip_assignment`?

Reply:
It's kind of referencing all of them, but that could definitely be made clearer.

Reply:
ffc0636